### PR TITLE
Add ConfigurationToConfigurations planner adapter.

### PR DESCRIPTION
**Pull request by brianhou**
This PR introduces the DART `ConfigurationToConfigurations` problem and planner. It also introduces an adapter from non-DART `ConfigurationToConfigurationPlanner` to DART `ConfigurationToConfigurationsPlanner`.

I think the ideal scenario would be to have two adapters instead:
- non-DART `ConfigurationToConfigurationPlanner` to non-DART `ConfigurationToConfigurationsPlanner`
- non-DART `ConfigurationToConfigurationsPlanner` to DART `ConfigurationToConfigurationsPlanner`

This isn't supported by the current `PlannerAdapter`, since the adapted planner must be a DART planner. However, I've also added the non-DART `ConfigurationToConfigurations` planner, which might eventually be useful for this purpose.

***

**Before creating a pull request**

- [x] Document new methods and classes
- [x] Format code with `make format`

**Before merging a pull request**

- [x] Set version target by selecting a milestone on the right side
- [x] Summarize this change in `CHANGELOG.md`
- [x] Add unit test(s) for this change

### CODE REVIEW

File: src/planner/dart/ConfigurationToConfiguration_to_ConfigurationToConfigurations.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  if (!configurationRanker)
+  {
+    auto nominalState = mMetaSkeletonStateSpace->createState();
+    mMetaSkeletonStateSpace->copyState(startState, nominalState);
+    configurationRanker = std::make_shared<const NominalConfigurationRanker>(
+        mMetaSkeletonStateSpace, mMetaSkeleton, nominalState);
+  }
+
+  std::vector<MetaSkeletonStateSpace::ScopedState> configurations;
+  for (const auto& goal : problem.getGoalStates())
```
Comment (sniyaz) [inline src/planner/dart/ConfigurationToConfiguration_to_ConfigurationToConfigurations.cpp:58]: Nit: Is there a reason to use a ref here? I think each `goal` is just a pointer/adress.

**Review by sniyaz — CHANGES_REQUESTED** [review]
Thanks so much for doing this! Just a few small issues: LMK when you're done, and I'll reformat and make sure Travis is happy before merging.

File: include/aikido/planner/dart/ConfigurationToConfiguration_to_ConfigurationToConfigurations.hpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  ConfigurationToConfiguration_to_ConfigurationToConfigurations(
+      std::shared_ptr<aikido::planner::ConfigurationToConfigurationPlanner>
+          planner,
+      ::dart::dynamics::MetaSkeletonPtr metaSkeleton,
+      distance::ConstConfigurationRankerPtr configurationRanker = nullptr);
+
+  // Documentation inherited.
+  virtual trajectory::TrajectoryPtr plan(
+      const ConfigurationToConfigurations& problem,
+      Planner::Result* result) override;
```
[the remark this answers is not in the document]
Comment (brianhou) [reply include/aikido/planner/dart/ConfigurationToConfiguration_to_ConfigurationToConfigurations.hpp:37]: That only seems to be the default in the VectorFieldPlanner, so I'm inclined to leave it as is for now.

[the remark this answers is not in the document]
Comment (sniyaz) [reply include/aikido/planner/dart/ConfigurationToConfiguration_to_ConfigurationToConfigurations.hpp:37]: Sounds good!

**Review by sniyaz — APPROVED** [review]
Thanks for the changes @brianhou! I fixed some really tiny things and ran make format, and this looks great.

Will merge once Travis looks good.

**Pull request merged**